sdks/python/apache_beam/io/parquetio_test.py

```diff
@@ -296,7 +296,9 @@ def test_sink_transform_int96(self):
               path, self.SCHEMA96, num_shards=1, shard_name_template='')
 
   def test_sink_transform(self):
-    with tempfile.NamedTemporaryFile() as dst:
+    dst = tempfile.NamedTemporaryFile()
```

Comment:
What is the reason for this change? (Is it because temp files cannot be re-opened in Windows?)

After dst.close() the file could be deleted anytime and that will make the test flaky. You can use tempfile.mkstemp() instead, which moves the cleanup requirement to the caller.

Reply:
+1, this pattern (here and elsewhere) looks strange to me.